Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -539,8 +539,9 @@ void OffscreenCanvas::commitToPlaceholderCanvas()
 
     // FIXME: Transfer texture over if we're using accelerated compositing
     if (m_context && (m_context->isWebGL() || m_context->isAccelerated())) {
-        m_context->prepareForDisplayWithPaint();
-        m_context->paintRenderingResultsToCanvas();
+        if (m_context->compositingResultsNeedUpdating())
+            m_context->prepareForDisplay();
```

Comment:
Should this logic be part of `m_context->drawBufferToCanvas`?

Reply:
No, the buffer swap logic is explicit, where as drawBufferToCanvas can just select which one to draw